Crash with Mixer Out of Process Adapter

**Describe the bug**
Crash when adapting Mixer Out of Process Adapter to use logentry template

**Expected behavior**
I expected my handler to receive a message

**Steps to reproduce the bug**
Follow the steps in https://github.com/istio/istio/wiki/Mixer-Out-of-Process-Adapter-Walkthrough

But use this operator config
```yaml
# handler for adapter mygrpcadapter
apiVersion: "config.istio.io/v1alpha2"
kind: handler
metadata:
  name: h1
  namespace: istio-system
spec:
  adapter: mygrpcadapter
  connection:
    address: "[::]:9000" #replaces at runtime by the test
  params:
    file_path: "out.txt"
---

# instance for template logentry
apiVersion: "config.istio.io/v1alpha2"
kind: instance
metadata:
  name: i1logentry
  namespace: istio-system
spec:
  template: logentry
severity: '"Default"'
timestamp: request.time
variables:
  requestSize: request.size | 0
---

# rule to dispatch to handler h1
apiVersion: "config.istio.io/v1alpha2"
kind: rule
metadata:
  name: r1
  namespace: istio-system
spec:
  actions:
  - handler: h1
    instances:
    - i1logentry
---
```

**Version**
$ istioctl version
Version: 1.0.2
GitRevision: d639408fded355fb906ef2a1f9e8ffddc24c3d64
User: root@66ce69d4a51e
Hub: gcr.io/istio-release
GolangVersion: go1.10.1
BuildStatus: Clean

Kubernetes version 1.10

**Installation**
I followed the steps here: https://istio.io/docs/setup/kubernetes/quick-start/

I chose option 1

**Environment**
MacOS 10.13.6
Also tested on Ubuntu 18.04, same problem

**Cluster state**
I am running minikube

https://gist.github.com/repenno/dc69bc8390065ecccf7d8aa2a1abfc2c

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x183bf4c]

goroutine 61 [running]:
istio.io/istio/mixer/pkg/runtime/routing.(*builder).getBuilderAndMapperDynamic.func1(0x1ff0d20, 0xc009575300, 0xc009575300, 0xc00010d601, 0xc00963dc20, 0x0)
	/root/go/src/istio.io/istio/mixer/pkg/runtime/routing/builder.go:523 +0x5c
istio.io/istio/mixer/pkg/runtime/dispatcher.(*session).dispatch(0xc00963db00, 0x1fef060, 0xc009637170)
	/root/go/src/istio.io/istio/mixer/pkg/runtime/dispatcher/session.go:164 +0x244
istio.io/istio/mixer/pkg/runtime/dispatcher.(*reporter).Report(0xc0096371d0, 0x1ff0d20, 0xc009575300, 0xc0004bb968, 0x2)
	/root/go/src/istio.io/istio/mixer/pkg/runtime/dispatcher/reporter.go:61 +0x8b
istio.io/istio/mixer/pkg/api.(*grpcServer).Report(0xc0004e8580, 0x1fef060, 0xc009637020, 0xc000444100, 0xc0004e8580, 0xc009636f30, 0x1a822e0)
	/root/go/src/istio.io/istio/mixer/pkg/api/grpcServer.go:281 +0x837
istio.io/istio/vendor/istio.io/api/mixer/v1._Mixer_Report_Handler(0x1ac9c80, 0xc0004e8580, 0x1fef060, 0xc009637020, 0xc009639a40, 0x0, 0x0, 0x0, 0xc000ae7c00, 0x0)
	/root/go/src/istio.io/istio/vendor/istio.io/api/mixer/v1/service.pb.go:106 +0x23e
istio.io/istio/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc000c3a480, 0x2001a00, 0xc009a28300, 0xc000ae7c00, 0xc00033ff50, 0x314eb78, 0x0, 0x0, 0x0)
	/root/go/src/istio.io/istio/vendor/google.golang.org/grpc/server.go:982 +0x4cd
istio.io/istio/vendor/google.golang.org/grpc.(*Server).handleStream(0xc000c3a480, 0x2001a00, 0xc009a28300, 0xc000ae7c00, 0x0)
	/root/go/src/istio.io/istio/vendor/google.golang.org/grpc/server.go:1208 +0x1308
istio.io/istio/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000a0d170, 0xc000c3a480, 0x2001a00, 0xc009a28300, 0xc000ae7c00)
	/root/go/src/istio.io/istio/vendor/google.golang.org/grpc/server.go:686 +0x9f
created by istio.io/istio/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
	/root/go/src/istio.io/istio/vendor/google.golang.org/grpc/server.go:684 +0xa1

--- a/mixer/pkg/runtime/config/config_test.go
+++ b/mixer/pkg/runtime/config/config_test.go
@@ -1952,6 +1952,38 @@ Attributes:
 		wantErr: "adapter='testCR1.adapter.default': unable to find template 'mymetric.default'",
 	},
 	{
+		Name: "missing params in instance",
+		Events1: []*store.Event{
+			updateEvent("mymetric.template.default", &adapter_model.Template{
+				Descriptor_: tmpl1Base64Str,
+			}),
+			updateEvent("testCR1.instance.default", &descriptorpb.Instance{
+				Template: "mymetric.default",
+			}),
+		},
+		E: `
+ID: 0
+TemplatesStatic:
+  Name: apa
+  Name: check
+  Name: quota
+  Name: report
+AdaptersStatic:
+  Name: adapter1
+  Name: adapter2
+HandlersStatic:
+InstancesStatic:
+Rules:
+TemplatesDynamic:
+  Resource Name:  mymetric.template.default
+    Name:  mymetric.template.default
+    InternalPackageDerivedName:  foo
+Attributes:
+  template.attr: BOOL
+`,
+		wantErr: "instance='testCR1.instance.default'.params: params cannot be nil",
+	},
+	{
 		Name: "delete template - break instance",
 		Events1: []*store.Event{
 			updateEvent("mymetric.template.default", &adapter_model.Template{
@@ -1959,6 +1991,7 @@ Attributes:
 			}),
 			updateEvent("testCR1.instance.default", &descriptorpb.Instance{
 				Template: "mymetric.default",
+				Params:   map[string]interface{}{},
 			}),
 			deleteEvent("mymetric.template.default"),
 		},

--- a/mixer/pkg/runtime/config/ephemeral.go
+++ b/mixer/pkg/runtime/config/ephemeral.go
@@ -324,6 +324,13 @@ func (e *Ephemeral) processDynamicInstanceConfigs(templates map[string]*Template
 
 		inst := resource.Spec.(*config.Instance)
 		instanceName := key.String()
+
+		if inst.Params == nil {
+			appendErr(errs, fmt.Sprintf("instance='%s'.params", instanceName),
+				counters.instanceConfigError, "params cannot be nil")
+			continue
+		}
+
 		log.Debugf("Processing incoming instance config: name='%s'\n%s", instanceName, resource.Spec)
 
 		tmpl, _ := getCanonicalRef(inst.Template, constant.TemplateKind, key.Namespace, func(n string) interface{} {
@@ -350,23 +357,22 @@ func (e *Ephemeral) processDynamicInstanceConfigs(templates map[string]*Template
 		var enc dynamic.Encoder
 		var params map[string]interface{}
 		var err error
-		if inst.Params != nil {
-			if _, ok := inst.Params.(map[string]interface{}); !ok {
-				appendErr(errs, fmt.Sprintf("instance='%s'.params", instanceName),
-					counters.instanceConfigError, "invalid params block. It must be of type map[string]interface{}")
-				continue
-			}
-			params = inst.Params.(map[string]interface{})
-			// name field is not provided by instance config author, instead it is added by Mixer into the request
-			// object that is passed to the adapter.
-			params["name"] = fmt.Sprintf("\"%s\"", instanceName)
-			enc, err = b.Build(getTemplatesMsgFullName(template.PackageName), params)
-			if err != nil {
-				appendErr(errs, fmt.Sprintf("instance='%s'.params", instanceName),
-					counters.instanceConfigError, "config does not conform to schema of template '%s': %v",
-					inst.Template, err.Error())
-				continue
-			}
+
+		if _, ok := inst.Params.(map[string]interface{}); !ok {
+			appendErr(errs, fmt.Sprintf("instance='%s'.params", instanceName),
+				counters.instanceConfigError, "invalid params block. It must be of type map[string]interface{}")
+			continue
+		}
+		params = inst.Params.(map[string]interface{})
+		// name field is not provided by instance config author, instead it is added by Mixer into the request
+		// object that is passed to the adapter.
+		params["name"] = fmt.Sprintf("\"%s\"", instanceName)
+		enc, err = b.Build(getTemplatesMsgFullName(template.PackageName), params)
+		if err != nil {
+			appendErr(errs, fmt.Sprintf("instance='%s'.params", instanceName),
+				counters.instanceConfigError, "config does not conform to schema of template '%s': %v",
+				inst.Template, err.Error())
+			continue
 		}
 
 		cfg := &InstanceDynamic{
@@ -721,7 +727,7 @@ func (e *Ephemeral) processDynamicTemplateConfigs(counters Counters, errs *multi
 		}
 
 		result[templateName] = &Template{
-			Name: templateName,
+			Name:                       templateName,
 			InternalPackageDerivedName: name,
 			FileDescSet:                fds,
 			PackageName:                desc.GetPackage(),

